### PR TITLE
docs: refactor doc structure

### PR DESCRIPTION
**Pull request by xiaofei-du**
Because

- The root README.md is too long and needs to be restructured

This commit

- replace `trigger mechanism` of a data source with `pipeline mode` concept and add the docs in `docs/pipeline-mode.md`.
- move `prepare-you-own-model` doc to `docs/model.md`
- add `check-yaml` in `pre-commit`
- update redoc service name to `redoc_openapi`

The `pipeline mode` is determined by the combination of data source and destination. It describes how an end-to-end pipeline processes its workload.

### CODE REVIEW

File: docs/pipeline-mode.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+### SYNC
+A pipeline with the `SYNC` mode responds to requests synchronously. The result is sent back to the user once the data is processed. This mode is a good fit for use cases with real-time requirements.
+<p align="center">
+<img src="mermaid/sync.svg" alt="Synchronous pipeline mode" />
+</p>
+
+To create a pipeline with the `SYNC` mode, please use `Direct` for both data source and destination.
+
+### ASYNC (Coming soon)
```
Comment (pinglin) [inline docs/pipeline-mode.md:14]:
```suggestion
### ASYNC (coming soon)
```

File: docs/pipeline-mode.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+A pipeline with the `ASYNC` mode responds to asynchronous workload. The user triggers the pipeline with an asynchronous request and only receives an acknowledged response. Once the data is processed, the result is sent to the data destination to be retrieved later. This mode is a good fit for use cases that do not need results immediately.
```
Comment (pinglin) [inline docs/pipeline-mode.md:16]:
```suggestion
A pipeline with the `ASYNC` mode responds to asynchronous workload. The user triggers the pipeline with an asynchronous request and only receives an acknowledged response. Once the data is processed, the result is sent to the data destination to be retrieved later. This mode is a good fit for use cases that do not need inference results immediately.
```

File: docs/pipeline-mode.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Pipeline Mode
+
+Depending on the requirements of your use case, you can create a pipeline with a suitable _Pipeline Mode_: synchronous (`SYNC`) and asynchronous (`ASYNC`).
+The mode of a pipeline is determined by the combination of data source and data destination. We use the diagrams below to show the difference.
+
+### SYNC
+A pipeline with the `SYNC` mode responds to requests synchronously. The result is sent back to the user once the data is processed. This mode is a good fit for use cases with real-time requirements.
```
Comment (pinglin) [inline docs/pipeline-mode.md:7]:
```suggestion
A pipeline with the `SYNC` mode responds to requests synchronously. The result is sent back to the user once the data is processed. This mode is a good fit for use cases with real-time inference requirements.
```

**Review by pinglin — CHANGES_REQUESTED** [review]

**Comment by xiaofei-du** [comment]
Thanks for the review. Check the latest commits for all suggested changes.

**Review by pinglin — APPROVED** [review]

**Comment by xiaofei-du** [comment]
@bochengyang, i will merge this PR.

**Pull request merged**